### ISSUE TITLE
Invalid time field in 'start' and 'end' doesn't throw error in MAM module.

### ISSUE THREAD

**Issue by shahdarshan1711** [processone/ejabberd#1520]
> What version of ejabberd are you using?

16.09

> What operating system (version) are you using?

CentOS 7.1

> How did you install ejabberd (source, package, distribution)?

Source

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behavior? What was the expected result?

When querying history data using XEP-313, giving 'start' value as `2017-01-09T09:60:50.358Z` doesn't throw bad request error. It should throw error as minute parameter should be between 00 to 59 according to XEP-82. If minute/second is greater than 60 it throws proper error "Bad-request". Only on corner scenario of minute/second = 60 and hour = 24 , it fails to validate.

Possible fix:

```
--- a/ejabberd/ejabberd-16.09/src/jlib.erl
+++ b/ejabberd/ejabberd-16.09/src/jlib.erl
@@ -763,8 +763,8 @@ parse_time1(Time) ->
           [Val] -> binary_to_integer(str:left(Val, 6, $0))
         end,
     [H, M, S] = str:tokens(HMS, <<":">>),
-    {[H1, M1, S1], true} = check_list([{H, 24}, {M, 60},
-                                      {S, 60}]),
+    {[H1, M1, S1], true} = check_list([{H, 23}, {M, 59},
+                                      {S, 59}]),
     {{H1, M1, S1}, MS}.
 
 check_list(List) ->
```

**Comment by zinid** [comment]
There is nothing to patch as jlib.erl is now deprecated.

**Comment by zinid** [comment]
I just tried via new interface:
```erlang
> xmpp_util:decode_timestamp(<<"2017-01-09T09:60:50.358Z">>).
** exception error: {bad_timestamp,<<"2017-01-09T09:60:50.358Z">>}
     in function  xmpp_util:decode_timestamp/1 (src/xmpp_util.erl, line 132)
```

So I consider this is fixed in 17.01.
